### ISSUE TITLE
[Bug]: Instantiate ATProtoKit and send request imitiatlly leads to a crash

### ISSUE THREAD

**Issue by mfreiwald** [MasterJ93/ATProtoKit#75]
### Summary

Creating a ATProtoKit instance, callign authenticate and getProfile leads to a crash

### Reproduction Steps

I have following code snippet which leads to a crash of the app (see screenshot below)
```swift
    static public func authenticate(handle: String, appPassword: String) async throws(ConnectError) -> (UserSession, AppBskyLexicon.Actor.ProfileViewDetailedDefinition) {
        let config = ATProtocolConfiguration(handle: handle, appPassword: appPassword)
        let protoClient = ATProtoKit(sessionConfiguration: config)

        do {
            try await config.authenticate()
        } catch {
            throw .notAuthenticated
        }

        guard let profile = try? await protoClient.getProfile(handle) else { throw .noSession }
        guard let session = config.session else { throw .noSession }
        return (session, profile)
    }
```

However, if I add a short delay after the ATProtoKit.init it works as expected.
There must be a problem with the unstructured Task in the ATProtoKit.init regarding the registry setup.

### Expected Results

No crash

### Actual Results

Crash of the app

### What operating systems did you experience this bug? (We'll count Docker as an operating system.)

macOS

### Operating System Version

macOS 15.2/iOS 18.2

### ATProtoKit Version

develop

### Additional Context

![CleanShot 2024-12-17 at 07 02 38@2x](https://github.com/user-attachments/assets/f005a671-1083-4c60-9581-186dc73017e8)

**Comment by MasterJ93** [comment]
Thanks for the report. I figured that there would be a bug like this but did think it would _actually_ get triggered. I've been meaning to fix that but other important tasks stood in the way.

I'll work on fixing that bug and it should arrive either in the next patch release or the one after, but your workaround that you've made should suffice for now (even though it's not ideal).

**Comment by MasterJ93** [comment]
I've created a temporary solution: the initializer will allow itself to complete once the code within the `Task` block is completed. This is also not an ideal solution as it can potentially block the main thread. Looking at your code, I don't exactly think this should be a problem, and no other developer has complained that their programs crash, so it feels like this is an edge case.

However, I don't like what I wrote as I don't want the class to block the thread, even if it's for less than 10 milliseconds (the longest I've seen the code run for; usually it's in microseconds or roughly 5 milliseconds).

I'll continue to work on fixing this issue, but for now, I will close this issue as it is resolved.
